internal/network/cidr_validations.go

```diff
@@ -125,3 +126,24 @@ func VerifyNetworkHostPrefix(prefix int64) error {
 	}
 	return nil
 }
+
+// Verify if the constrains for dual-stack machine networks are met:
+//   * there are exactly two machine networks
+//   * the first one is IPv4 subnet
+//   * the second one is IPv6 subnet
+func VerifyMachineNetworksDualStack(networks []*models.MachineNetwork, isDualStack bool) error {
+	if !isDualStack {
+		return nil
+	}
+	if len(networks) != 2 {
+		return errors.Errorf("Expected 2 machine networks, found %d", len(networks))
+	}
+	if !IsIPV4CIDR(string(networks[0].Cidr)) {
+		return errors.Errorf("First machine network has to be IPv4 subnet")
+	}
+	if !IsIPv6CIDR(string(networks[1].Cidr)) {
+		return errors.Errorf("Second machine network has to be IPv6 subnet")
```

Comment:
It'd be nice (in a follow-up PR, perhaps) to add the Cidr in the error message

Reply:
Yeah, I'd prefer to add it as an improvement later on